test/ServiceFabric.Mocks.NetCoreTests/SerializationTests/MockReliableDictionaryTests.cs

```diff
@@ -0,0 +1,133 @@
+﻿using System;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using System.Threading.Tasks;
+using ServiceFabric.Mocks.ReliableCollections;
+using System.Threading;
+
+namespace ServiceFabric.Mocks.NetCoreTests.SerializationTests
+{
+    [TestClass]
+    public class MockReliableDictionaryTests
+    {
+        const string originalContentValue = "original value";
+        const string modifiedContentValue = "modified value";
+
+        /// <summary>
+        /// The value is stored as serialized object, so changes made after storing it are not visible in the queried result.
+        /// </summary>
+        /// <returns></returns>
+        [TestMethod]
+        public async Task DictionarySerializedValueChangesIgnoredTest()
+        {
+            const string key = "key";
+
+            var value = new ModifyablePayload
+            { 
+                Content = originalContentValue
+            };
+
+            SerializerCollection serializers = new();
```

Comment:
What about testing the real path - calling `TryAddStateSerializer()` followed by `GetOrAddAysnc` to create the dictionary?